src/osm_configurator/view/popups/yes_no_pop_up.py

```diff
@@ -22,6 +23,37 @@ def __init__(self, message, func):
 
         Args:
             message (str): The message to be shown in the PopUp.
-            func (typing.Callable): A Function that takes one Boolean and has no return, for the PopUp to send a message back.
+            func (typing.Callable): Function taking one Boolean and has no return for the PopUp to return a message
         """
-        pass
+        self.func = func
+
+        popup = super().__init__()
+        self.geometry(POPUPSIZE)
+
+        self.title("Alert")
+
+        customtkinter.CTkLabel(popup, text=message) \
+            .pack(side="top", fill="both", expand="True", padx=10, pady=10)
+
+        customtkinter.CTkButton(self, text="Accept", command=combine_funcs(accept, self.destroy)) \
+            .pack(side="bottom", padx=10, pady=10)
+
+        self.button = customtkinter.CTkButton(self, text="Cancel", command=combine_funcs(cancel, self.destroy)) \
+            .pack(side="bottom", padx=10, pady=10)
+
+    def accept(self):
+        global gloBla
+        self.func(True)
+
+    def cancel(self):
+        self.func(False)
+
+
+def combine_funcs(*funcs):
```

Comment:
method out of class? why?

Reply:
The Popup is a non-callable object during execution. We therefore cannot execute methods calling other methods. Having the function inside the class would lead to an error.

Reply:
ok